op-batcher/batcher/channel_manager.go

```diff
@@ -60,6 +63,7 @@ func (s *channelManager) Clear() {
 	s.log.Trace("clearing channel manager state")
 	s.blocks = s.blocks[:0]
 	s.tip = common.Hash{}
+	s.closed = false
```

Comment:
If I see correctly, it currently wouldn't happen that we clear the channel manager after closing it, because we don't load L2 blocks any more, which is where the `Clear()` might be called. 

But if we ever get into this situation, I think we should actually keep the `channelManager` closed forever once it got closed. The batcher doesn't revert from shutting down.

```suggestion
// We deliberately let `s.closed` stay at its current value, because once a channelManger got closed, it should stay closed.

Reply:
We're intending to stop and potentially restart the batcher via the [Admin API](https://github.com/ethereum-optimism/optimism/blob/develop/op-batcher/rpc/api.go), and resetting this `closed` flag would be necessary to support a restart.

Reply:
I see, didn't think about restarting a stopped batcher in the same program instance yet. I assumed that when it is stopped, the whole application would be restarted.

I think quite a few more changes would need to be made to support restarting the batcher using the same `BatchSubmitter` instance.  E.g. the lifetime context `shutdownCtx` gets closed etc. Since the batcher is supposed to be stateless anyways, it is probably easier to just always restart the whole application. Or at least create a new `BatchSubmitter` instance instead of reusing it.

But anyways, I'm fine with also resetting `c.closed` to false here, since it doesn't have any affect currently (as far as I can tell). We can rack our brains about how to best restart the batch submitter via rpc when the time comes.

Reply:
@sebastianst Base has a stop/start batcher flow in our runbook, which we use every now-and-then. The `shutdownCtx` and `killCtx` gets recreated on `start` so it should work fine. I agree that it's relatively easy to restart the entire process, but for us it's been easier to call an RPC to reset the state, because our environment is pretty locked down for security reasons.

Reply:
Sounds good. We could also consider changing the rpc `Start` implementation to throw away a shut-down `BatchSubmitter` instance (i.e., if stopped before) and re-create a new one. This way, the restart would still happen inside the same process but we wouldn't have the headache of making sure that a `BatchSubmitter` instance can safely be reused after shutdown. It would probably come with its own problems though.